src/sentry/api/endpoints/organization_events_trace.py

```diff
@@ -204,8 +205,61 @@ def child_sort_key(item: TraceEvent) -> List[int]:
 
 
 def query_trace_data(
-    trace_id: str, params: Mapping[str, str]
+    trace_id: str, params: Mapping[str, str], use_snql: bool = False
 ) -> Tuple[Sequence[SnubaTransaction], Sequence[SnubaError]]:
+    if use_snql:
+        transaction_query = QueryBuilder(
+            Dataset.Discover,
+            params,
+            query=f"trace:{trace_id}",
+            selected_columns=[
+                "id",
+                "transaction.status",
+                "transaction.op",
+                "transaction.duration",
+                "transaction",
+                "timestamp",
+                "project",
+                "project.id",
```

Comment:
Does `project.id` no longer get added automatically? Are there other places that depend on this automatic insertion of `project.id`? If so, should that be fixed instead of adding it to `selected_columns` manually?

Reply:
Not anymore, it was added automatically cause there was no way to do the project transform without adding project.id.